"Contributing"

arokem opened the pull request:
This file gets rendered by Github into PRs (https://github.com/blog/1184-contributing-guidelines) and should help newcomers figure out what/how/why to contribute. For the git procedure, I am directing them to the existing developer documentation, but this also covers some other points that we like to look for in our review. Am I forgetting anything important?

jchoude commented on CONTRIBUTING:
```diff
@@ -0,0 +1,86 @@
+### Documentation
+
+We use [sphinx](http://www.sphinx-doc.org/en/stable/contents.html) for
+documentation. We welocme contributions of examples, and suggestions for
```
Typo in welcome

jchoude commented on CONTRIBUTING:
```diff
@@ -0,0 +1,86 @@
+documentation. We welocme contributions of examples, and suggestions for
+changes in the documentation, but please make sure that changes that are
+introduced render properly into the html format that is used for the Dipy
+website
```
Missing the closing period.

jchoude commented on doc/devel/python3.rst:
```diff
@@ -25,22 +28,22 @@ As the first code line of the file, to use Python 3 behavior by default.
 
-Yeah, you knew that, but use the ``__future__`` import above, and
-``print(something)``
+In Python 3, ``print`` is  function. Please use the ``__future__`` import above,
```
Missing an "a" between "is" and "function"?

jchoude commented:
Seems like a good addition. Very nice to know that they now render this automatically!

arokem commented:
Thanks for taking a look! I fixed that typos you identified and also added
a sentence about the numpy docstring standard, which I forgot to mention
previously.

On Mon, Jan 18, 2016 at 7:01 AM, Jean-Christophe Houde <
notifications@github.com> wrote:

> Seems like a good addition. Very nice to know that they now render this
> automatically!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/838#issuecomment-172553000>.
>

Garyfallidis commented on CONTRIBUTING:
```diff
@@ -0,0 +1,89 @@
+Dipy is an open-source software project, and we have an open development
+process. This means that we welcome contributions from anyone. We do ask that
+you first read this document and follow the guidelines we have outlined here and
+that you follow the [NIPY community code of conduct](http://nipy.org/conduct.html).
```
Is this conduct inspired from somewhere or written from scratch?

arokem replied:
As it says, inspired by others and adapted for our needs. See https://github.com/nipy/nipy.github.com/pull/1

Garyfallidis commented on CONTRIBUTING:
```diff
@@ -0,0 +1,89 @@
+### Supporting both Python 2 and 3
+
+Most of the functionality in Dipy works on both Python 3 and Python 2. Please
+follow the instructions [here](http://nipy.org/dipy/devel/python3.html) to
```
You can write http://dipy.org/devel/...
We got the domain so we can make the links shorter. However, it's a bit puzzling that when someone writes dipy.org/devel we will get nipy.org/dipy/devel in his browser but maybe there is a way to fix this (from godaddy).

Garyfallidis commented on CONTRIBUTING:
```diff
@@ -0,0 +1,89 @@
+# Contributing to Dipy
```
Thanx for starting to write such a document. Shouldn't this be an rst file in the dipy/doc folder? 
Here you wrote a markdown file but we don't have any markdown documents in dipy. I have a feeling this is in the wrong format and possibly folder. This information should  be rendered by sphinx or something in the docs. You can also have a copy in the setup.py folder but that is secondary.

arokem replied:
Fixed that other things as well. Thanks for taking a look!

On Mon, Jan 18, 2016 at 11:31 AM, Eleftherios Garyfallidis <
notifications@github.com> wrote:

> In CONTRIBUTING
> <https://github.com/nipy/dipy/pull/838#discussion_r50035912>:
>
> > @@ -0,0 +1,89 @@
> > +# Contributing to Dipy
>
> Thanx for starting to write such a document. Shouldn't this be an rst file
> in the dipy/doc folder?
> Here you wrote a markdown file but we don't have any markdown documents in
> dipy. I have a feeling this is in the wrong format and possibly folder.
> This information should be rendered by sphinx or something in the docs. You
> can also have a copy in the setup.py folder but that is secondary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/838/files#r50035912>.
>

arokem commented:
It's a markdown file, and at top level, so that it interoperates well with Github. See here: https://github.com/blog/1184-contributing-guidelines. TLDR: If there's a markdown file called `CONTRIBUTING.md` (or `CONTRIBUTING`) at top level, people get pointed to this file when they make a PR.
